File: python/others/typical90_bc.py
n,p,q=map(int,input().split())
a=[*map(int,input().split())]
ans=0
for i in range(n):
    for j in range(i+1,n):
        for k in range(j+1,n):
            for l in range(k+1,n):
                for m in range(l+1,n):
                    s=1
                    s*=a[i];s%=p
                    s*=a[j];s%=p
                    s*=a[k];s%=p
                    s*=a[l];s%=p
                    s*=a[m];s%=p
                    if s==q:ans+=1
print(ans)

File: rust/typical90/src/bin/typical90_bc.rs
fn main(){
    proconio::input!{
        (n,p,q):(usize,usize,usize),
        a:[usize;n]
    }
    let mut ans = 0;
    for i in 0..n{
        for j in i+1..n{
            for k in j+1..n{
                for l in k+1..n{
                    for m in l+1..n{
                        let s = a[i]%p*a[j]%p*a[k]%p*a[l]%p*a[m]%p;
                        if s==q { ans+=1 }
                    }
                }
            }
        }
    }
    println!("{}",ans);
}